### PR TITLE
Add quickstart template for Azure Jenkins to Container Registry

### PR DESCRIPTION
**Pull request by clguimanMSFT**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [ x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@clguimanMSFT__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (clguiman). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by clguimanMSFT** [comment]
@rcarun @EricJizbaMSFT please have a look

File: 201-docker-image-to-azure-container-registry-using-jenkins/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+<a href="https://aka.ms/azdeployacrpipelineonubuntu" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="https://aka.ms/azvisualizeacrpipelineonubuntu" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+The template allows you to host an instance of Jenkins on a DS1_v2 size Linux Ubuntu 14.04 LTS VM in Azure. It will also create an Azure Container Registry and return the full registry URL.
+
+You can optionally include a basic Jenkins pipeline that will checkout a user-provided git repository with a Dockerfile embedded and it will build an push the Docker container in the provisioned Azure Container Registry.
```
Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/README.md]: right

File: 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      "type": "string",
+      "value": "[concat('http://', reference(variables('publicIPAddressName')).dnsSettings.fqdn,':8080')]"
+    },
+    "SSH": {
+      "type": "string",
+      "value": "[concat('ssh ', parameters('adminUsername'), '@', reference(variables('publicIPAddressName')).dnsSettings.fqdn)]"
+    },
+    "Azure Container Registry URL": {
+      "type": "string",
+      "value": "[concat('http://',reference(resourceId('Microsoft.ContainerRegistry/registries', variables('acrName'))).loginServer)]"
```
Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.json]: The docker push command in Jenkins requires the http (https)

File: 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.parameters.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "adminUsername": {
+      "value": "GEN-USERNAME"
+    },
+    "adminPassword": {
+      "value": "GEN-PASSWORD"
+    },
+    "jenkinsDnsLabelPrefix": {
+      "value": "GEN-UNIQUE"
+    },
+    "vmSize": {
```
Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.parameters.json]: it was at some point :) I'll remove it

File: 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[variables('acrStorageAccountName')]",
+      "apiVersion": "2016-01-01",
+      "location": "[resourceGroup().location]",
+      "sku": {
+        "name": "Standard_LRS"
+      },
+      "kind": "Storage",
+      "properties": {}
```
Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/azuredeploy.json]: this is the VM's storage account. I just changed the ACR storage account definition to be encrypted.

File: 201-docker-image-to-azure-container-registry-using-jenkins/metadata.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+  "itemDisplayName": "Docker Image to Azure Container Registry using Jenkins",
```
Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/metadata.json]: @rcarun please review

Comment (rcarun) [inline 201-docker-image-to-azure-container-registry-using-jenkins/metadata.json]: Please change to "Jenkins to Azure Container Registry"

Comment (clguimanMSFT) [inline 201-docker-image-to-azure-container-registry-using-jenkins/metadata.json]: done

**Pull request merged**